app/components/CameraComponent.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+import {
+  Alert,
+  Dimensions,
+  Image,
+  StyleSheet,
+  View,
+} from 'react-native';
+import Camera from 'react-native-camera';
+import Button from 'react-native-button';
+
+export default class CameraComponent extends Component {
+
+  state = {
+    image: null,
+  };
+
+  _takePicture() {
+    const setState = this.setState.bind(this);
+    this.camera.capture()
+      .then((data) => {
+        this.setState({image: data.path});
+      })
+      .catch(err => {
+        Alert.alert(
+          'An error occurred.',
+          err,
+          [
+            {
+              text: 'OK',
+              onPress: () => { setState({image: null}); }
+            },
+          ],
+        );
+        console.error(err);
+      });
+  }
+
+  render() {
+    const { image } = this.state;
+    if (image != null) {
+      return (
+        <View
+          style={styles.container}>
+          <Image
+            style={styles.camera}
+            source={{uri: image}}/>
+            <View
```

Comment:
This is indented one level too deep - it looks like it's a child of the `<Image />` component